litestar/typing.py

```diff
@@ -96,7 +96,7 @@ def _parse_metadata(value: Any, is_sequence_container: bool, extra: dict[str, An
     example_list: list[Any] | None
     if example := extra.pop("example", None):
         example_list = [Example(value=example)]
-    elif examples := getattr(value, "examples", None):
+    elif examples := (extra.pop("examples", None) or getattr(value, "examples", None)):
```

Comment:
Gives examples in json_schema_extra precedence over `Field(examples=...)`.

Reply:
Nice! I understood that my fix was a kind of desperate thing in the wrong place (because the typing in the surrounding code was already incorrect) but I would have never found this.

For the sake of making sense of the types, may I ask why there is a cast to `list[str]` in the next line? I understand that we are now converting the raw examples to `list[Example]`, but I don't understand the assumption (built into the cast) that all examples are `str`? Couldn't they be anything?
```
example_list = [Example(value=example) for example in cast("list[str]", examples)]
```

Reply:
I can't really shed any light on why that is - but it seems unnecessary and arbitrary. `Example.value` is typed `Any` so the `cast()` to `str` for the values seems redundant, and `examples` in that context would be `Any` also due to `dict.pop()` and `getattr()` to assign its value, so the cast to list wouldn't be necessary either I don't think...